extensions/VisualEditor/wikia/modules/ve/init/ve.init.wikia.Optimizely.js

```diff
@@ -0,0 +1,29 @@
+/* global toolbarABTestVariantNumber:false */
+/*!
+ * Utility functions for running Optimizely A/B tests with VisualEditor
+ */
+
+/**
+ * @returns {number} The A/B test variant number for the Toolbar A/B test
+ */
+ve.init.wikia.getToolbarABTestVariantNumber = function () {
+	// toolbarABTestVariantNumber is set for variant #1 for all Optimizely toolbar experiments
+	if ( typeof ( toolbarABTestVariantNumber ) === 'undefined' ) {
+		return 0;
+	}
+
+	return toolbarABTestVariantNumber;
+};
+
+/**
+ * Hook for activating experiment on VE load
+ */
+mw.hook( 've.activate' ).add( function () {
+	// Production tests are split by language. This is done because each language has a separate engagement percentage
+	var optimizelyIds = window.wgDevelEnvironment ?
+		[ '4721410313' ] : [ '4701112678', '5003080344', '5003533755', '5013830116' ];
+
+	optimizelyIds.forEach( function (element) {
+		window.optimizely.push( [ 'activate', element ] );
```

Comment:
optimizely.push:
According to what I see in the Optimizely documentation for optimizely.push: "will activate any running manual experiments whose targeting conditions are satisfied by the current visitor on this page"

So unless I am misunderstanding, this should be ok. It is of course possible to check for window.location.href and then only call optimizely.push for the matching one, but I thought I could avoid that.

There is no change in the issue for GA reporting, so reporting is the same as before, e.g. dialog-image-insert, dialog-video-insert.